clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/AbstractEcsCachingAgent.java

```diff
@@ -212,11 +213,13 @@ CacheResult buildCacheResult(
   protected boolean keyAccountRegionFilter(String authoritativeKeyName, String key) {
     Map<String, String> keyParts = Keys.parse(key);
     return keyParts != null
-        && keyParts.get("account").equals(accountName)
-        &&
-        // IAM role keys are not region specific, so it will be true. The region will be checked of
-        // other keys.
-        (authoritativeKeyName.equals(IAM_ROLE.ns) || keyParts.get("region").equals(region));
+        && ((accountName.equals(keyParts.get("account"))
+                &&
+                // IAM role keys are not region specific, so it will be true. The region will be
```

Comment:
nit: this comment doesn't quite make sense now that the check includes application cache agent too

Reply:
updated